### ISSUE TITLE
Portal authentication service

### ISSUE THREAD

**Issue by anapolg** [sonata-nfv/tng-portal#5]
Coming from [#245](https://git.cs.upb.de/5gtango/UserStories/issues/245). Username/email and password will be authenticated with oauth2.

**Comment by jbonnet** [comment]
Hi, @anapolg, we'd like to (re-)use this [authentication API](https://github.com/sonata-nfv/son-gkeeper/wiki/Authentication-usage). OAuth2 would be great to use (e.g., GitHub's, GitLab's or BitBucket's for developers, Google's ro Facebook's for customers), but it's a bit tricky to use for authenticating in APIs... what do you think?

**Comment by anapolg** [comment]
Ok I did not know about that API. We'll use that one. Shoud I close this issue then?

**Comment by jbonnet** [comment]
Not necessarily...
That implementation is a little clumsy (me to blame!): it depends on Keycloak, a rather heavy solution. I'd prefer the lightness of an OAuth2 solution, but we have also 'systems' (i.e., not humans, like the OSS systems of a Telco) as users of the API, which usually prefer a `API KEY` for authentication (see this [Aha! Oauth2 Authentication](https://www.aha.io/api/oauth2))...

For March, there's no time in re-doing this (we'll re-use waht we have), but we can leave this opened, just not to forget about all this.

**Comment by anapolg** [comment]
Yes, perfect. Should I conserve then [this](https://git.cs.upb.de/5gtango/generalInfo/wikis/5gtango-portal) page of the wiki too? I began to describe the process with oauth2 and talk about kwt tokens. We were thinking to use them to allow authentication as well as to grant permissions to the user, all that codified in the JWT token.

**Comment by jbonnet** [comment]
Yes, please keep it...
